File: Python/August_Problems/problem860_15th_august_2024.py
from typing import List


def lemonadeChange(bills: List[int]) -> bool:
    coins = [0, 0]
    for bill in bills:
        if bill == 5:
            coins[0] += 1
        elif bill == 10:
            if coins[0] > 0:
                coins[0] -= 1
                coins[1] += 1
            else:
                return False
        else:
            if coins[1] > 0 and coins[0] > 0:
                coins[1] -= 1
                coins[0] -= 1
            elif coins[0] >= 3:
                coins[0] -= 3
            else:
                return False
        return True


def problem860():
    pass


if __name__ == "__main__":
    problem860()

File: CPP/Year-2024/August_Problems/problem860_15th_august_2024.cpp
#include <iostream>
#include <vector>

using namespace std;

bool lemonadeChange(vector<int> &bills)
{
    int change_set[3] = {0, 0, 0};
    for (int i = 0; i < bills.size(); i += 1)
    {
        if (bills[i] == 5)
        {
            change_set[0] += 1;
        }
        else if (bills[i] == 10)
        {
            if (change_set[0] > 0)
            {
                change_set[0] -= 1;
                change_set[1] += 1;
            }
            else
            {
                return false;
            }
        }
        else
        {
            if (change_set[1] >= 1 and change_set[0] >= 1)
            {
                change_set[0] -= 1;
                change_set[1] -= 1;
                change_set[2] += 1;
            }
            else
            {
                if (change_set[1] == 0)
                {
                    if (change_set[0] >= 3)
                    {
                        change_set[2] += 1;
                        change_set[0] -= 3;
                    }
                    else
                    {
                        return false;
                    }
                }
                else
                {
                    return false;
                }
            }
        }
    }
    return true;
}

int main()
{
    return 0;
}